### ISSUE TITLE
'Didn't find op for builtin opcode ...' error

### ISSUE THREAD

**Issue by expeon07** [eloquentarduino/EloquentTinyML#43]
Hi! I'm trying to upload a TFLite model that uses autoencoder with tanh activation function in the layers but I get the following errors when uploading

```
Start
GetModel done
Version check done
Didn't find op for builtin opcode 'TANH' version '1'

Failed to get registration from op code  d
 
Allocation failed
AllocateTensors() failed
```

Is there any special configuration that I should do or tanh is not supported yet?

**Comment by eloquentarduino** [comment]
Tanh seems to be supported: https://github.com/tensorflow/tflite-micro/blob/main/tensorflow/lite/micro/all_ops_resolver.cc
Do you have any chance to replace tanh with something else to test if the rest of network works?

**Comment by expeon07** [comment]
Yes, I substituted it with sigmoid and relu and they didn't get an error.

**Comment by eloquentarduino** [comment]
As far as I know, to implement RNN you should use GRU instead of LSTM.
GRU should be supported.

**Comment by expeon07** [comment]
I tried substituting the LSTM layer with the GRU as well, but it gave the same error.
I also closely followed this implementation here: 
https://colab.research.google.com/github/tensorflow/tensorflow/blob/master/tensorflow/lite/examples/experimental_new_converter/Keras_LSTM_fusion_Codelab.ipynb#scrollTo=tB1NZBUHDogR
but to no avail. :(

**Comment by eloquentarduino** [comment]
I will take a look at LSTM/GRU this week and see if I find something that
works. Be patient.

Il sab 26 mar 2022, 22:44 Expeon ***@***.***> ha scritto:

**Comment by expeon07** [comment]
Thanks a lot! That will be very helpful.

**Comment by eloquentarduino** [comment]
LSTM is indeed supported as many users reported.
Be sure to use Tf==2.2.3
